programs/test/benchmark.c

```diff
@@ -93,7 +93,7 @@ int main( void )
 /*
```

Comment:
ASan reports memory leaks in the ECDH tests with Everest enabled. I haven't investigated where, but given that it leaks 76114 allocations, it must be inside the library and not just in the benchmark. Steps to reproduce:

```
scripts/config.pl set MBEDTLS_ECDH_VARIANT_EVEREST_ENABLED
mkdir build-asan
cd build-asan
cmake -D CMAKE_BUILD_TYPE:String=ASanDbg ..
make
programs/test/benchmark ecdh
```

Reply:
Finally found them! These leaks were due to the fact that, in the new benchmark (full handshake), the client context was re-initialized for each handshake when using the non-Everest context. I blindly assumed that, if I have to call `mbedtls_ecdh_setup` for the server, I also have to do so for the client. However, the client calls `mbedtls_ecdh_setup` via `mbedtls_ecdh_read_params`. I don't like this asymmetry, so I fixed it by allowing clean re-initialization, see 2340f03c597b923c0f427c76b4c3d2cd11638410. Alternatively, we can also move the call to `mbedtls_ecdh_free` for the client context into the `TIME_PUBLIC` loop of the benchmark. @yanesca is that how you intended the new API to work, do you want to remove `mbedtls_ecdh_setup` from `mbedtls_ecdh_read_params`, or are you happy with the clean re-initialization?

Reply:
(`test_suite_ecdh` now fails because my fix wasn't conservative enough it seems, I'll fix that in the meanwhile.)

Reply:
(Fixed now 4a43d14146220e8550d6ad87cb798f74ce9ee209)

Reply:
This behaviour of `mbedtls_ecdh_setup()` is intentional:
https://github.com/ARMmbed/mbedtls/blob/9c99dc862cd5cf9cdc62cbfcd200063c83295397/include/mbedtls/ecdh.h#L227-L229

Originally this was done by accessing the context members directly and the goal of introducing `mbedtls_ecdh_setup()` was to provide an abstraction.

I would prefer to fix the benchmark application instead of adding support for re-initialisation.

Reply:
Done! 3327dc02743993e7dd3cdaf2c0bc4407206058c6 (I'll remove the reverted commits at the next rebase)